src/client/eppo-client.ts

```diff
@@ -351,6 +358,90 @@ export default class EppoClient implements IEppoClient {
+    let variation = this.getStringAssignment(flagKey, subjectKey, subjectAttributes, defaultValue);
+    let action: string | null = null;
+    const banditKey = variation;
+    try {
+      const banditParameters = this.banditConfigurationStore?.get(banditKey);
+      if (banditParameters && !Object.keys(actions ?? {}).length) {
+        // If it's a bandit, but we have no actions, just return default value
+        variation = defaultValue;
+      } else if (banditParameters) {
+        // For now, we use the shortcut of assuming if a variation value is the key of a known bandit, that is the bandit we want
+        const banditModelData = banditParameters.modelData;
+        const banditEvaluation = this.banditEvaluator.evaluateBandit(
+          flagKey,
+          subjectKey,
+          subjectAttributes,
+          actions,
+          banditModelData,
+        );
+        action = banditEvaluation.actionKey;
+
+        const banditEvent: IBanditEvent = {
+          timestamp: new Date().toISOString(),
+          featureFlag: flagKey,
+          bandit: banditKey,
+          subject: subjectKey,
+          action,
+          actionProbability: banditEvaluation.actionWeight,
+          optimalityGap: banditEvaluation.optimalityGap,
+          modelVersion: banditParameters.modelVersion,
+          // TODO: bucket these out ahead of time
+          subjectNumericAttributes: this.pruneValuesByType(subjectAttributes, true),
+          subjectCategoricalAttributes: this.pruneValuesByType(subjectAttributes, false),
+          actionNumericAttributes: this.pruneValuesByType(actions[action], true),
+          actionCategoricalAttributes: this.pruneValuesByType(actions[action], false),
+          metaData: this.buildLoggerMetadata(),
+        };
+        this.logBanditAction(banditEvent);
+      }
+    } catch (err) {
+      if (this.isGracefulFailureMode) {
+        logger.error('Error evaluating bandit action', err);
+        variation = defaultValue;
+      } else {
+        throw err;
+      }
+    }
+    return { variation, action };
+  }
+
+  private logBanditAction(banditEvent: IBanditEvent): void {
+    if (!this.banditLogger) {
+      // No bandit logger set; enqueue the event in case a logger is later set
+      if (this.queuedBanditEvents.length < MAX_EVENT_QUEUE_SIZE) {
+        this.queuedBanditEvents.push(banditEvent);
+      }
+      return;
+    }
+    // If here, we have a logger
+    try {
+      this.banditLogger.logBanditAction(banditEvent);
+    } catch (err) {
+      logger.warn('Error encountered logging bandit action', err);
+    }
+  }
+
+  // TODO: this method can be repurposed to bucket attributes once bandit signatures updated
+  private pruneValuesByType(attributes: Attributes, numeric: boolean): Attributes {
+    const result: Attributes = {};
+    Object.entries(attributes).forEach(([key, value]) => {
+      const isNumeric = typeof value === 'number' && isFinite(value);
```

Comment:
nit: could be extracted as a separate function

Reply:
True, but this is presently only used in this one spot for, so thinking of keeping it inlined for now.